### PR TITLE
Better tracking of dependence in installation

### PR DESCRIPTION
**Pull request by izgzhen**

### CODE REVIEW

File: src/Rules/Install.hs [hunk 1]
```diff
@@ -131,18 +128,18 @@ withLatestBuildStage pkg m = do
...
 installPackageConf :: Action ()
-installPackageConf = do
+installPackageConf = do    
     let context = vanillaContext Stage0 rts
     liftIO $ IO.createDirectoryIfMissing True (takeDirectory pkgConfInstallPath)
     build $ Target context HsCpp [ pkgPath rts -/- "package.conf.in" ]
                                  [ pkgConfInstallPath <.> "raw" ]
-    Stdout out <- cmd ("grep" :: String) [ "-v", "^#pragma GCC"
-                                         , pkgConfInstallPath <.> "raw" ]
+    Stdout content <- cmd ("grep" :: String) [ "-v", "^#pragma GCC"
```
Comment (snowleopard) [inline src/Rules/Install.hs]: Do you need `:: String` because of `OverloadedStrings`? If yes, how do you use the extension? (I couldn't find.)

**Comment by snowleopard** [comment]
Looks good! Could you clarify what this improves, specifically?

File: src/Rules/Install.hs (hunk 1 above)
Comment (izgzhen) [inline src/Rules/Install.hs]: Thanks for points this out! It is not useful now, and I'll drop it. IIRC, it was initially for doing binary patch, so I need to write a literal `ByteString`.

**Comment by izgzhen** [comment]
> Could you clarify what this improves, specifically?

Sorry I didn't mention that it was for one point we discussed in #352: with this, we can do `build install` safely without doing `build` beforehand.

**Comment by snowleopard** [comment]
> we can do `build install` safely without doing `build` beforehand.

Great! I was hoping for this.

I'll merge when CI bots complete.

**Comment by izgzhen** [comment]
CI Completed

**Pull request merged**